src/pages/expenses.tsx

```diff
@@ -12,33 +11,111 @@ import {
   DrawerFooter,
   DrawerTrigger,
 } from "@/components/ui/drawer";
+import { useToast } from "@/components/ui/use-toast";
 
 export default function Expenses() {
-  const [expenses, setExpenses] = useState<Expense[]>(expensesData);
   const [isDrawerOpen, setIsDrawerOpen] = useState(false);
+  const { toast } = useToast();
 
-  const updateExpense = (updatedExpense: Expense) => {
-    setExpenses((prevExpenses) =>
-      prevExpenses.map((expense) =>
-        expense.id === updatedExpense.id ? updatedExpense : expense,
-      ),
-    );
+  const { data: expenses, refetch: refetchExpenses } =
+    api.expenses.getAll.useQuery();
+  const { data: totalExpenses } = api.expenses.getTotalExpenses.useQuery();
+
+  const addExpenseMutation = api.expenses.create.useMutation({
```

Comment:
This system of using the handlers onSuccess to handle any callbacks is fantastic and what we use at HQ. The alternative is to use a useEffect, which gets messy really quick. The fewer useEffects you can use, the better.